resteasy-client-api/pom.xml

```diff
@@ -45,6 +45,21 @@
             <artifactId>jakarta.json</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.microprofile.config</groupId>
+            <artifactId>microprofile-config-api</artifactId>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.smallrye.config</groupId>
+            <artifactId>smallrye-config</artifactId>
+            <exclusions>
+               <exclusion>
+                  <groupId>javax.annotation</groupId>
+                  <artifactId>javax.annotation-api</artifactId>
+               </exclusion>
+            </exclusions>
+        </dependency>

```

Comment:
This seems a little weird adding it to the API.

Reply:
Well, it's related to your previous comment, since WebApplicationExceptionWrapper uses MP Config:
```
    static WebApplicationException wrap(final WebApplicationException e) {
        final Config config = ConfigProvider.getConfig();
        final boolean originalBehavior = config.getOptionalValue("resteasy.original.webapplicationexception.behavior", boolean.class).orElse(false);
```

Reply:
Yeah. I think we should find a way to keep this part in the implementation. It feels like implementation details are leaking into the API here.

Reply:
I'm not sure I'm following. "resteasy.original.webapplicationexception.behavior" is meant to be available to the user ...

Reply:
In fact, maybe all of those user facing parameter names should be in resteasy-core-spi ... ?

Reply:
The parameter is fine, but the way we retrieve is the implementation detail right? For example using the Config API is an implementation detail of where the property is stored.

Reply:
Ah, I didn't see this when I just responded to your email. I guess in my mind, MP Config is like the US dollar ... the global standard. But, you're right, maybe next year will be using NanoProfile Config. ;-)

Reply:
Ha these days you never know :)